MUI/src/main/java/mui/MUISetupProvider.java

```diff
@@ -43,61 +53,110 @@ private void setLogProvider(MUILogProvider log) {
+				inputRow.setMinimumSize(new Dimension(800, 100));
+				GridBagConstraints inputRowConstraints = new GridBagConstraints();
+				inputRowConstraints.fill = GridBagConstraints.HORIZONTAL;
+
+				inputRowConstraints.gridx = 0;
+				inputRowConstraints.gridwidth = 3;
+				inputRowConstraints.gridy = 0;
+				inputRowConstraints.gridheight = 1;
+				inputRowConstraints.weightx = 0.75;
+
+				entry.setText((String) prop.get("default"));
+				inputRow.add(entry, inputRowConstraints);
+
+				JFileChooser chooser = new JFileChooser();
+				chooser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+				chooser.setDialogTitle("Set Workspace Folder");
+
+				JButton selectButton = new JButton("Select...");
+				selectButton.addActionListener(new ActionListener() {
+
+					@Override
+					public void actionPerformed(ActionEvent e) {
+						int returnVal = chooser.showOpenDialog(null);
+						if (returnVal == JFileChooser.APPROVE_OPTION) {
+							try {
+								String path = chooser.getSelectedFile().getCanonicalPath();
+								entry.setText(path);
+							}
+							catch (IOException e1) {
+								e1.printStackTrace();
+							}
+						}
+					}
 
-		mainPanelConstraints = new GridBagConstraints();
-		mainPanelConstraints.fill = GridBagConstraints.BOTH;
-		mainPanelConstraints.gridwidth = GridBagConstraints.REMAINDER;
-		mainPanelConstraints.weightx = 0.9;
-		mainPanelConstraints.weighty = 0.9;
+				});
 
-		inputPanel = new JPanel(new GridBagLayout());
+				inputRowConstraints.gridx = 3;
+				inputRowConstraints.gridwidth = 1;
+				inputRowConstraints.weightx = 0.25;
+				inputRow.add(selectButton, inputRowConstraints);
 
-		inputPanelConstraints = new GridBagConstraints();
-		inputPanelConstraints.fill = GridBagConstraints.BOTH;
+				formPanel.add(inputRow);
 
-		inputPanelConstraints.gridx = 0;
-		inputPanelConstraints.gridy = 0;
-		inputPanelConstraints.weightx = 0.25;
-		inputPanelConstraints.gridwidth = 1;
-		inputPanel.add(new JLabel("Program Path:"), inputPanelConstraints);
+			}
+			else if (prop.get("type") == "string" || prop.get("type") == "number") {
+				entry.setText(prop.get("default").toString());
+				entry.setToolTipText("Only 1 value allowed");
+				formPanel.add(entry);
+			}
+			else if (prop.get("type") == "array") {
+				// TODO: doesn't handle default param for arrays, but not needed as part of sensible defaults for running manticore on native binaries
+				// for now, same UI as string/num, and we will parse space-separated args
+				entry.setText(prop.get("default").toString());
+				entry.setToolTipText("You can space-separate multiple arguments");
+				formPanel.add(entry);
+			}
+			else {
+				// TODO: to achieve parity with Binja MUI, type==boolean must be supported, but not needed as part of sensible defaults for running manticore on native binaries
+				throw new UnsupportedOperationException(
+					String.format("[ERROR] Cannot create input row for %s with the type %s", name,
+						prop.get("type")));
+			}
```

Comment:
Would probably be easier to read if you created functions for each of these property types